src/network/networkd-dhcp-server.c

```diff
@@ -402,6 +412,30 @@ int dhcp4_server_configure(Link *link) {
         return 0;
 }
 
+int config_parse_dhcp_server_relay_agent_suboption(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char **suboption_value = data;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (!startswith(rvalue, "string:"))
+                return -EINVAL;
+
+        return free_and_strdup(suboption_value, rvalue + strlen("string:"));
```

Comment:
1. Please accept empty string and `"string:"`.
2. do not return negative errno on invalid input, but log about refusing the value and return 0.
3. `startswith()` returns pointer.

So,
```
if (isempty(rvalue)) {
        *suboption_value = mfree(*suboption_value);
        return 0;
}

p = startswith(rvalue, "string:");

if (!p) {
        log_syntax(...);
        return 0;
}

return free_and_strdup(suboption_value, empty_to_null(p));
```